test/controllers/nonsipp/BasicDetailsCheckYourAnswersControllerSpec.scala

```diff
@@ -350,6 +354,30 @@ class BasicDetailsCheckYourAnswersControllerSpec
       }
     }
 
+    "audit event" - {
+      "should only send audit event when CIP_START_EVENT_FLAG is true" in {
+        val userAnswers = currentTaxYearUserAnswersWithFewMembers
+        val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+
+        running(application) {
+          mockTaxYear(currentReturnTaxYear)
+          MockPsrSubmissionService.submitPsrDetails()
+
+          val request = FakeRequest(GET, onSubmit.url).withSession(CIP_START_EVENT_FLAG -> "true")
+          val result = route(application, request).value
+          status(result) mustEqual SEE_OTHER
+          verify(mockAuditService, times(1)).sendEvent(any())(using any(), any())
+          reset(mockAuditService)
+
+          val requestWithoutFlag = FakeRequest(GET, onSubmit.url)
```

Comment:
Very minor: there might be two tests here:
1. "should send audit event when CIP_START_EVENT_FLAG is true"
2. "should not send audit event when CIP_START_EVENT_FLAG is not present/defined"